prov/efa/src/efa_user_info.c

```diff
@@ -325,35 +325,13 @@ int efa_user_info_alter_rxr(struct fi_info *info, const struct fi_info *hints)
 		 * which means FI_MR_HMEM implies FI_MR_LOCAL for device buffer.
 		 */
 		if (hints->caps & FI_HMEM) {
-			if (ofi_hmem_p2p_disabled()) {
-				EFA_WARN(FI_LOG_CORE,
-					"FI_HMEM capability currently requires peer to peer support, which is disabled.\n");
-				return -FI_ENODATA;
-			}
-			//TODO: remove the rdma checks once FI_HMEM w/o p2p is supported
-
-			if (!efa_device_support_rdma_read()) {
-				EFA_WARN(FI_LOG_CORE,
-				        "FI_HMEM capability requires RDMA, which this device does not support.\n");
-				return -FI_ENODATA;
-
-			}
-
-			if (!rxr_env.use_device_rdma) {
-				EFA_WARN(FI_LOG_CORE,
-				        "FI_HMEM capability requires RDMA, which is turned off. You can turn it on by set environment variable FI_EFA_USE_DEVICE_RDMA to 1.\n");
-				return -FI_ENODATA;
-			}
-
```

Comment:
This depend on the value of `cuda_get_xfer_setting()`.

If `cuda_get_xfer_setting()` is CUDA_XFER_ENABLED. we support FI_HMEM.
Otherwise, we support FI_HMEM with p2p + read.

Reply:
Same issue here, if we restrict `FI_HMEM_CUDA` to just `FI_HMEM_CUDA_ENABLE_XFER=1`, there's no GDRCopy path since #8357 restricts `cuda_gdrcopy_dev_register` to the following conditions `efa_mr_hmem_setup()`:

```c
static inline
bool efa_mr_use_gdrcopy(struct efa_mr *efa_mr)
{
	return efa_mr->peer.iface == FI_HMEM_CUDA && cuda_get_xfer_setting()!=CUDA_XFER_ENABLED && cuda_is_gdrcopy_enabled();
}
```

Reply:
Yes. that is intented. we do not want to use gdrcopy to support HMEM on non-p2p systems. (performance is an issue, always data consistency is another).

Reply:
I see. This is news to me, but not a big deal. I can easily add the restriction on `p2p_required_by_impl` to `cuda_get_xfer_setting() == CUDA_XFER_ENABLED`.

Reply:
So which of the original conditions should be kept in? `!efa_device_support_rdma_read()`? E.g.

```c
if (!efa_device_support_rdma_read() && cuda_get_xfer_setting() != CUDA_XFER_ENABLED) {
    EFA_WARN(...);
    return -FI_ENODATA;
}
```